src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RSABCrypt.cs

```diff
@@ -20,6 +20,7 @@ internal sealed class RSABCrypt : RSA
 
         private SafeBCryptKeyHandle? _key;
         private int _lastKeySize;
+        private bool _publicOnly;
```

Comment:
Interesting... since RSABCrypt never has to use an existing handle it always knows if it has a private key or not.

Reply:
Yep. But I added the parameter to `SetKey` since that is the front door for setting the key handle. If we ever get to a point of needing to take a handle, SetKey will force callers to supply the information.